File: hileusesimple/src/main/java/com/xiongtao/hileusesimple/simple3/MainActivity.java
package com.xiongtao.hileusesimple.simple3;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.xiongtao.hileusesimple.R;
import com.xiongtao.hileusesimple.simple3.interfacedi.TestInterface;

import javax.inject.Inject;

import dagger.hilt.android.AndroidEntryPoint;


@AndroidEntryPoint  // 我是被注解的 被注入的
public class MainActivity extends AppCompatActivity {
    @Inject
    TestInterface testInterface;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // 为什么能够控制  字节码插庄
        // DaggerCOm。create.inject(this);
        findViewById(R.id.button).setOnClickListener(v -> {
            testInterface.method();
        });
    }
}

File: livedata_demo/src/main/java/com/xiongtao/livedata_demo/simple4/MainActivity.kt
package com.xiongtao.livedata_demo.simple4

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.xiongtao.livedata_demo.R

/**
 * @Description:
 * @Author: xiongtao
 * @Date: 2023-03-07 15:47
 */
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)

        // 如果LiveData默认的黏性，会产生Bug，你就需要剔除黏性

        // livedata发消息通知所有的观察者数据变化了
        OKLiveDataBusKT.with("data1",String::class.java,false).value = "sdfasdfa"
        OKLiveDataBusJava.getInstance().with("data2", String::class.java).value= "Nine Suns" // Java版本
        // 点击事件，跳转下一个Activity
        val button = findViewById<Button>(R.id.button)
        button.setOnClickListener {
            startActivity(Intent(this, MainActivity2::class.java))
        }
    }
}